pkg/asset/manifests/azure/cluster.go

```diff
@@ -48,16 +47,28 @@ func GenerateClusterAssets(installConfig *installconfig.InstallConfig, clusterID
 	computeSubnet := installConfig.Config.Platform.Azure.ComputeSubnetName(clusterID.InfraID)
 	networkSecurityGroup := installConfig.Config.Platform.Azure.NetworkSecurityGroupName(clusterID.InfraID)
 
+	// Create Control Plane Outbound Load Balancer.
+	// Set the Outbound Load Balancer type to `Public` if APIServer needs to be public.
+	lbType := capz.Internal
+	if installConfig.Config.PublicAPI() {
+		lbType = capz.Public
+	}
```

Comment:
Hm, I could very well be wrong, but I'm not sure we want to do this. Do you know what the effect of this is? 

The default `lbType` for ControlPlaneOutboundLB is `Public`. I believe even for Private API clusters we still want the outbound LB to be public, i.e. have a public IP address associated with it, because that is how we provide egress, but we do not want to route traffic through the LB to the api server (so that the api server still remains inaccessible on the public internet)...

Overall, what we are doing is using CAPZ to provision the ControlPlaneOutboundLB and then we subsequently modify it to turn it into a full public API server:

https://github.com/openshift/installer/blob/5ddaa22b836479367bb00b9fb3c0d43e924b6ee8/pkg/infrastructure/azure/azure.go#L517

This architecture and what we're doing is pretty complex, so I could definitely be missing something.

Reply:
I didn't find any documentation here https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/api/v1beta1/types.go#L356 regarding the default LBType.

Working on an update.